### PR TITLE
Update expense

### PR DESCRIPTION
**Pull request by LeoGrouet**

### CODE REVIEW

File: App/src/Service/ExpenseBalancer.php [hunk 1]
```diff
@@ -7,13 +7,14 @@
...
 use Doctrine\ORM\EntityManagerInterface;
 
 class ExpenseBalancer
 {
     public function __construct(
         private readonly EntityManagerInterface $entityManager,
-        private readonly UserBalanceRepository $userBalanceRepository
+        private readonly UserBalanceRepository $userBalanceRepository,
     ) {
     }
```
Comment (clementvtrd) [inline App/src/Service/ExpenseBalancer.php]: When all your attributes are readonly you can make the entire class readonly. 😉 Except those which extends a non-readonly class

**Review by clementvtrd — APPROVED** [review]

**Review by clementvtrd — CHANGES_REQUESTED** [review]
Wrong button 😭

File: App/src/Service/ExpenseBalancer.php [hunk 2]
```diff
@@ -10,13 +10,12 @@
...
-class ExpenseBalancer
+readonly class ExpenseBalancer
 {
     public function __construct(
         private readonly EntityManagerInterface $entityManager,
-        private readonly UserBalanceRepository $userBalanceRepository,
-    ) {
-    }
+        private readonly UserBalanceRepository $userBalanceRepository
+    ) {}
```
Comment (clementvtrd) [inline App/src/Service/ExpenseBalancer.php]:
```suggestion
final readonly class ExpenseBalancer
{
    public function __construct(
        private EntityManagerInterface $entityManager,
        private UserBalanceRepository $userBalanceRepository
    ) {}
```

Comment (clementvtrd) [reply App/src/Service/ExpenseBalancer.php]: Presque !

**Review by clementvtrd — APPROVED** [review]

File: App/src/Service/ExpenseBalancer.php (hunk 2 above)
Comment (LeoGrouet) [reply App/src/Service/ExpenseBalancer.php]: Final ? C'est juste une notation ou ça a un intérêt particulier le final

**Pull request merged**